modin/backends/pandas/query_compiler.py

```diff
@@ -1180,6 +1180,52 @@ def sort_index(self, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def melt(self, *args, **kwargs):
+        def _convert_to_list(x):
+            if is_list_like(x):
+                x = [*x]
+            elif x is not None:
+                x = [x]
+            else:
+                x = []
+            return x
+
+        row_lengths = self._modin_frame._row_lengths
+
+        id_vars, values_vars = map(
+            _convert_to_list,
+            [kwargs.get("id_vars", None), kwargs.get("value_vars", None)],
+        )
+
+        value_vars = len(values_vars)
+        if value_vars == 0:
+            value_vars = len(self.columns) - len(id_vars)
+
+        shuffled = self._modin_frame._apply_full_axis(
```

Comment:
I understand why you chose `_apply_full_axis` at first. Do you think it is possible to use `_apply_select_indices`? We will need to potentially replicate the untouched partitions. It will also improve performance significantly.

Reply:
It could be implemented via `_apply_select_indices` but then we will have to broadcast partitions of `id_vars` columns to every partition, which is not so fast since we have multicolumn `id_vars`, for example if `modin_frame` with `id_vars` columns has more than one column partitions we can't just do that shiny broadcast like we do in [`groupby_reduce`](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/partition_manager.py#L78-L97), in that case we gotta use approach from [`binary_operation`](https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/partition_manager.py#L635-L672) which is actually applying function at the full axis.

Nevertheless, even if we could implement that with `_apply_select_indices` that will not solve the problem with slow reordering after applying `melt`. Any thoughts about reordering?

Reply:
Since `melt` is made for wide data frames, I'll try to reimplement it with `_apply_full_axis_select_indices(0, ...)`. If initially we have wide DataFrame with a lot of column partitions, so it should scale pretty well if we'll apply function at columns axis.